kotlinx-coroutines-core/jvm/src/debug/internal/DebugProbesImpl.kt

```diff
@@ -183,10 +187,27 @@ internal object DebugProbesImpl {
         dumpCoroutinesSynchronized(out)
     }
 
+    /*
+     * Filters out coroutines that do not call probeCoroutineCompleted,
+     * are completed, but not yet garbage collected.
+     *
+     * Typically, we intercept completion of the coroutine so it invokes "probeCoroutineCompleted",
+     * but it's not the case for lazy coroutines that get cancelled before start.
+     */
+    private fun CoroutineOwner<*>.notFinished(): Boolean {
+        // Guarded by lock
+        val job = info.context?.get(Job) ?: return true
```

Comment:
Seems okay to skip coroutines with garbage-collected context (will be filtered one call later) or without a job

Reply:
It looks Ok. However, to my taste double-negation in the logic makes it hard to understand. I'd rather call it `isFinished` an negate its return value.